src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -81,141 +76,73 @@ public ExamAccessService(ExamRepository examRepository, ExerciseGroupRepository
 
         // Check that the exam is visible
         if (exam.getVisibleDate() != null && exam.getVisibleDate().isAfter(ZonedDateTime.now())) {
-            return forbidden();
+            return badRequest("exam.visibleDate", "400", "Can't access the exam before it is visible!");
         }
 
-        return ResponseEntity.ok(studentExam.get());
-    }
-
-    /**
-     * Checks if the current user is allowed to manage exams of the given course.
-     *
-     * @param courseId The id of the course
-     * @param <T>      The type of the return type of the requesting route so that the response can be returned there
-     * @return an optional with a typed ResponseEntity. If it is empty all checks passed
-     */
-    public <T> Optional<ResponseEntity<T>> checkCourseAccessForEditor(Long courseId) {
-        Course course = courseRepository.findByIdElseThrow(courseId);
-        if (!authorizationCheckService.isAtLeastEditorInCourse(course, null)) {
-            return Optional.of(forbidden());
-        }
-        return Optional.empty();
+        return ResponseEntity.ok(studentExam);
     }
 
     /**
-     * Checks if the current user is allowed to manage exams of the given course.
+     * Checks if the current user is at least of role in the given course, otherwise throws an AccessForbiddenException
      *
+     * @param role the role to check the access for in the course
      * @param courseId The id of the course
-     * @param <T>      The type of the return type of the requesting route so that the response can be returned there
-     * @return an optional with a typed ResponseEntity. If it is empty all checks passed
-     */
-    public <T> Optional<ResponseEntity<T>> checkCourseAccessForInstructor(Long courseId) {
+     * @throws AccessForbiddenException if user isn't at least of role in course
+     * */
+    public void checkCourseAccessForRoleElseThrow(Role role, Long courseId) throws AccessForbiddenException {
         Course course = courseRepository.findByIdElseThrow(courseId);
-        if (!authorizationCheckService.isAtLeastInstructorInCourse(course, null)) {
-            return Optional.of(forbidden());
+        switch (role) {
```

Comment:
Why is the `role` unwrapped with the switch and then reconstructed for method call inside? Why is it not possible to replace the whole switch with a
`authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(role, course, null);` here?